src/index.js

```diff
@@ -162,9 +160,12 @@ function makeQuery(options) {
   const tryFetchQueryPages = async pageVariables => {
     try {
       // Perform the query
-      const data = await Promise.all(
-        pageVariables.map(variables => query.queryFn(variables))
-      )
+      const promises = pageVariables.map(variables => query.queryFn(variables))
+
+      query.cancelQueries = () =>
```

Comment:
The only place to access the user provided `cancel` method on the promise returned from `queryFn` is here, and because this code will execute before `query.promise` is defined, the method cannot be stored as `query.promise.cancel`.

Reply:
👏 👏 👏